proposals/debug-id.md

```diff
@@ -138,58 +138,42 @@ with the following variables:
+## Polyfills
 
-In the absence of browser support for loading debug IDs of particular stack frames, code transformers can inject a snippet into _all_ individual generated JavaScript files to maintain a global dictionary that maps from resource URL to Debug ID:
+For this proposal, we include a [repository for "polyfilling" Debug IDs](https://github.com/getsentry/javascript-debug-ids).
+It includes an implementation of plugins for various popular build-tooling as well as an implementation for a runtime API to access Debug IDs.
 
-```javascript
-(function () {
-  var stack = new Error().stack; // stack is non-standard and may be undefined
-  var match = stack && stack.match(/(?:\bat |@)(.*?):\d+:\d+$/m);
-  if (match) {
-    var ids = (globalThis.__DEBUG_IDS__ = globalThis.__DEBUG_IDS__ || {});
-    ids[match[1]] = "<DEBUG_ID>";
-  }
-})();
-```
-
-```javascript
-function getDebugIdForUrl(url) {
-  return globalThis.__DEBUG_IDS__ && globalThis.__DEBUG_IDS__[url];
-}
-```
+Note: While polyfilling is possible and is in wide production use already[^1], we have found a plethora of issues:
+- Complexity in setup and compatibility
+- Polyfills usually require nasty workarounds for build-tool quirks
+- Build-tools often don't allow for modifying source-maps
+- Injecting Debug IDs into transitive dependencies is error prone and in some cases ruins the entire polyfilling process
+- The polyfills inflate bundle-size more than necessary
+- Chicken-and-egg situations with [Subresource Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity)
 
-## Appendix D: Parsing Debug IDs
+## Implementors
 
-The following Python code shows how Debug IDs are to be extracted from generated JavaScript and source map files:
+The following Source Map **Generators** have implemented Debug IDs as proposed:
 
-```python
-import re
-import uuid
-import json
+- Rollup ([`output.sourcemapDebugIds` option](https://rollupjs.org/configuration-options/#output-sourcemapdebugids))
+- Oxc ([`debug_id` API](https://docs.rs/oxc/latest/oxc/sourcemap/struct.JSONSourceMap.html#structfield.debug_id))
+- Expo ([Injected by default](https://docs.expo.dev/versions/latest/config/metro/#source-map-debug-id))
+- Rolldown ([`output.sourcemapDebugIds` option](https://github.com/rolldown/rolldown/pull/2516))
 
+The following Source Map **Consumers/Debuggers** have implemented Debug IDs:
 
-_debug_id_re = re.compile(r'^//# debugId=(.*)')
+- Sentry.io ([Docs](https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/artifact-bundles/#artifact-bundles))
 
+The following implementations are work-in-progress:
+- **Generator:** Webpack ([PR to extend `devtool` option](https://github.com/webpack/webpack/pull/18947))
+- **Generator:** Turbopack (underlying `sourcemap` Rust crate has been updated)
+- **Consumer:** V8 ([CL to extend `Error.prepareStackTrace`](https://chromium-review.googlesource.com/c/v8/v8/+/5979833), is awaiting for proposal to reach stage 3)
```

Comment:
I'd rather not have a work-in-progress section, since it can easily get out of date.

Reply:
Removed the section 👍